### PR TITLE
Revolver Stuff

### PR DESCRIPTION
**Pull request by ZomgPonies**
- Makes the cap revolver an actual revolver child. Yay less snowflake
- Gives the "Spin cylinder" verb to all revolvers.

### CODE REVIEW

**Comment by ZomgPonies** [comment]
Shit, need a map edit to replace the cap guns on map

**Comment by Plotron** [comment]
Yay, spinnin' dem all cylinders.

**Comment by TheDZD** [comment]
Cap revolvers exist on maps, so you're getting mapping errors.

**Comment by ZomgPonies** [comment]
Yes I know so Im waiting on shuttle merge first.

File: code/modules/projectiles/guns/projectile/revolver.dm [hunk 1]
```diff
@@ -64,8 +64,26 @@
...
+	update_icon()
+
+/obj/item/weapon/gun/projectile/revolver/capgun
+	name = "cap gun"
+	desc = "Looks almost like the real thing! Ages 8 and up."
+	origin_tech = "combat=1;materials=1"
+	mag_type = "/obj/item/ammo_box/magazine/internal/cylinder/cap"
+
 /obj/item/weapon/gun/projectile/revolver/detective
-	name = "\improper .38 Mars Special"
```
Comment (Fox-McCloud) [inline code/modules/projectiles/guns/projectile/revolver.dm]: Why? This isn't a .357 revolver. Put this name back, please.

**Comment by Fox-McCloud** [comment]
This should be able to be updated now, map wise.

File: code/modules/projectiles/projectile/bullets.dm [hunk 2]
```diff
@@ -8,6 +8,17 @@
...
+/obj/item/projectile/bullet/cap
+	name = "cap"
+	damage = 0
+	nodamage = 1
+	embed = 0
+	sharp = 0
+
+/obj/item/projectile/bullet/cap/process()
+	loc = null
+	qdel(src)
```
Comment (Fox-McCloud) [inline code/modules/projectiles/projectile/bullets.dm]: del()

projectiles don't play nice with qdel().

**Comment by Iamgoofball** [comment]
pretty good/10

**Comment by ZomgPonies** [comment]
Should be ready for merge.

File: code/modules/projectiles/guns/projectile/revolver.dm (hunk 1 above)
Comment (Fox-McCloud) [inline code/modules/projectiles/guns/projectile/revolver.dm]: This still needs addressed.

**Comment by Fox-McCloud** [comment]
@ZomgPonies 

The detective's revolver name needs to stay--once that's fixed, this should be merge ready.

**Comment by Fox-McCloud** [comment]
@ZomgPonies ? Want to merge this in; just have to fix that little goof with the detective's revolver and I can.

**Comment by CrAzYPiLoT-SS13** [comment]
Uh, what's happening with this?

**Comment by Plotron** [comment]
@CrAzYPiLoT-SS13 
Zomg is taking long to respond, most likely.

**Comment by CrAzYPiLoT-SS13** [comment]
Wow, 38 days. What's the record?

File: code/modules/projectiles/guns/projectile/revolver.dm [hunk 3]
```diff
@@ -168,43 +186,11 @@
 	desc = "A Russian-made revolver for drinking games. Uses .357 ammo, and has a mechanism that spins the chamber before each trigger pull."
 	origin_tech = "combat=2;materials=2"
 	mag_type = "/obj/item/ammo_box/magazine/internal/cylinder/rus357"
-	var/spun = 0
+
 
 /obj/item/weapon/gun/projectile/revolver/russian/New()
 	..()
 	Spin()
```
Comment (Tastyfish) [inline code/modules/projectiles/guns/projectile/revolver.dm]: This is causing runtimes by the way, `Spin()` is now a verb that expects a `usr` to be set.

**Pull request merged**